__tests__/src/integration.spec.js

```diff
@@ -20,9 +20,34 @@ expect.extend({ toMatchImageSnapshot });
 
 describe('integration tests', () => {
   const imagePath = path.resolve(__dirname, '../stubs', 'TestImage.png');
-  const data = fs.readFileSync(imagePath);
+  const imageData = fs.readFileSync(imagePath);
+  const snapPath = './__tests__/src/__image_snapshots__/integration-snap.png';
+
+  beforeAll(() => {
+    if (fs.exists(snapPath)) {
+      fs.unlink(snapPath);
+    }
+  });
+
+  it('writes a snapshot with no error.', () => {
+    expect(() => expect(imageData).toMatchImageSnapshot({ customSnapshotIdentifier: 'integration' })).not.toThrowError();
+  });
 
   it('matches an identical snapshot.', () => {
-    expect(data).toMatchImageSnapshot();
+    expect(() => expect(imageData).toMatchImageSnapshot({ customSnapshotIdentifier: 'integration' })).not.toThrowError();
+  });
+
+  it('fails with a different snapshot.', () => {
+    const failImagePath = path.resolve(__dirname, '../stubs', 'TestImageFailure.png');
+    const failImageData = fs.readFileSync(failImagePath);
+
+    expect(() => expect(failImageData).toMatchImageSnapshot({ customSnapshotIdentifier: 'integration' })).toThrowError();
```

Comment:
Can you use `toThrowErrorMatchingSnapshot()` to be more specific about the error expected to be thrown? And same in the other test.